UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FssService.cs

```diff
@@ -27,40 +28,97 @@ public class FssService : IFssService
+                if (batchStatusResponse.IsSuccessStatusCode)
+                {
+                    _logger.LogInformation(EventIds.GetBatchStatusRequestCompleted.ToEventId(), "Request to get batch status for BatchID - {BatchID} completed | Batch Status is {BatchStatus} | {DateTime} | StatusCode : {StatusCode} | _X-Correlation-ID : {CorrelationId}", batchId, batchStatus, DateTime.Now.ToUniversalTime(), batchStatusResponse.StatusCode.ToString(), CommonHelper.CorrelationID);
+
+                    FssBatchStatusResponseModel fssBatchStatusResponseModel = JsonConvert.DeserializeObject<FssBatchStatusResponseModel>(await batchStatusResponse.Content.ReadAsStringAsync());
+                    Enum.TryParse(fssBatchStatusResponseModel?.Status, false, out batchStatus);
 
-                if (!batchStatusResponse.IsSuccessStatusCode)
+                    if (breakBatchStatus.Contains(batchStatus))
+                    {
+                        _logger.LogInformation(EventIds.FssBatchStatusPollingStopped.ToEventId(), "Polling to FSS to get batch status for BatchID - {BatchID} stopped | Batch Status is {BatchStatus} | {DateTime} | _X-Correlation-ID : {CorrelationId}", batchId, batchStatus, DateTime.Now.ToUniversalTime(), CommonHelper.CorrelationID);
+                        break;
+                    }
+                    await Task.Delay(int.Parse(_fssApiConfiguration.Value.BatchStatusPollingDelayTime));
+                }
+                else
                 {
-                    _logger.LogError(EventIds.BatchStatusRequestFailed.ToEventId(), "Request to get batch status for BatchID - {BatchId} failed at {DateTime} | StatusCode:{StatusCode} | _X-Correlation-ID:{CorrelationId}", batchId, batchStatusResponse.StatusCode.ToString(), DateTime.Now.ToUniversalTime(), CommonHelper.CorrelationID);
-                    break;
+                    _logger.LogError(EventIds.GetBatchStatusRequestFailed.ToEventId(), "Request to get batch status for BatchID - {BatchID} failed | {DateTime} | StatusCode : {StatusCode} | _X-Correlation-ID : {CorrelationId}", batchId, DateTime.Now.ToUniversalTime(), batchStatusResponse.StatusCode.ToString(), CommonHelper.CorrelationID);
+                    throw new FulfilmentException(EventIds.GetBatchStatusRequestFailed.ToEventId());
                 }
-                FssBatchStatusResponseModel responseObj = JsonConvert.DeserializeObject<FssBatchStatusResponseModel>(await batchStatusResponse.Content.ReadAsStringAsync());
-
-                Enum.TryParse(responseObj?.Status, false, out batchStatus);
+            }
 
-                _logger.LogInformation(EventIds.BatchStatusRequestCompleted.ToEventId(), "Request to get batch status for BatchID - {BatchId} completed. Batch Status is {BatchStatus} at {DateTime} | _X-Correlation-ID:{CorrelationId}", batchId, batchStatus, DateTime.Now.ToUniversalTime(), CommonHelper.CorrelationID);
+            if (batchStatus != FssBatchStatus.Committed)
```

Comment:
This if statement and log message are no longer accurate given the loop will now be exited early for other statuses. Please adjust to take into account the other statuses

Reply:
Resolved